src/main/java/com/hedera/hashgraph/sdk/HederaReceiptStatusException.java

```diff
@@ -0,0 +1,42 @@
+package com.hedera.hashgraph.sdk;
+
+import com.hedera.hashgraph.proto.ResponseCodeEnum;
+import com.hedera.hashgraph.proto.TransactionGetReceiptQuery;
+import com.hedera.hashgraph.proto.TransactionGetReceiptResponse;
+
+/**
+ * A {@link HederaStatusException}, thrown on error status by {@link TransactionId#getReceipt(Client)}.
+ *
+ * The receipt is included, though only the {@link TransactionReceipt#status} field will be
+ * initialized; all the getters should throw.
+ */
+public class HederaReceiptStatusException extends HederaStatusException {
+    /**
+     * The ID of the transaction that failed, in case that context is no longer available
+     * (e.g. the exception was bubbled up).
+     */
+    public final TransactionId transactionId;
+
+    /**
+     * The receipt of the transaction that failed; the only initialized field is
+     * {@link TransactionReceipt#status}.
+     */
+    public final TransactionReceipt receipt;
+
+    HederaReceiptStatusException(ResponseCodeEnum responseCode, TransactionId transactionId, TransactionReceipt receipt) {
+        super(responseCode);
+        this.transactionId = transactionId;
+        this.receipt = receipt;
+    }
+
```

Comment:
Needs getMessage override so it will display the transactionID and the responsecode.